### PR TITLE
Feature reviews

### PR DESCRIPTION
**Pull request by shinichikudo000**
Added Review, Comment, and Like models along with their respective controllers.
Introduced is_completed boolean and review_notification attribute in the Appointment model.
Adjusted associations in User, Appointment, Review, Comment, and Like models for consistency.
Resolved seeding issues to ensure data integrity.

Issues to be Fixed:

Adjusted migration file order so that Appointment migrations precede Review migrations for logical consistency.
Added JavaScript controllers and their corresponding view files; these files should be ignored for now as they still need to be fixed for views.

To-do
Views and stream for likes, comments and forms
pagination(maybe by scroll), search review
image

image

### CODE REVIEW

**Review by p3rc1us — APPROVED** [review]

File: app/assets/images/communication.png
Comment (cimafrancamae) [inline app/assets/images/communication.png:1]: can we use icons for these images instead? we have font-awesome installed: https://fontawesome.com/icons

File: app/components/shadcn/select_component.rb
Comment (cimafrancamae) [inline app/components/shadcn/select_component.rb:1]: this component is being used by other forms, have you confirmed that this doesn't affect the others?

File: config/routes.rb
Comment (cimafrancamae) [inline config/routes.rb:1]: Have you merged dev here? The collection under `resources :services` is gone

**Comment by cimafrancamae** [comment]
Is there anything I need to configure locally?
I ran `rails db:seed` and it returned an error:

image


Then `rails assets:precompile` returned this error:

image

File: config/routes.rb
Comment (cimafrancamae) [reply config/routes.rb:1]: please don't remove this on the routes:

```
resources :services do
    collection do
      post :category, to: 'categories#select', as: :select_category_new
      post :set_detail, to: 'service_details#set', as: :set_detail
      get :new_form, to: 'service_details#show', as: :detail
    end
  end
```

File: app/assets/images/communication.png
Comment (shinichikudo000) [reply app/assets/images/communication.png:1]: naka svg siya. ill try to use na lang icons. nakalimutan lang idelete yung files ng png

File: app/components/shadcn/select_component.rb
Comment (shinichikudo000) [reply app/components/shadcn/select_component.rb:1]: not sure wala siya sa merging issues ko sa vscode. ininstall ko lang siya so yung akin is from shadcn. ill fix na lang

File: config/routes.rb
Comment (shinichikudo000) [reply config/routes.rb:1]: nag merge ako sa dev. and wala namang akong finix sa routes. not sure bakit wala

**Review by cimafrancamae — APPROVED** [review]
This PR is already working on my end! 🚀

**Review by TyJacalan — CHANGES_REQUESTED** [review]
Generally good commit, lots of new features and we can now use Redis throughout the application. Checklist before approving this PR:
1. revert file: [select_helper](app/helpers/components/select_helper.rb)
2. revert file: [service reference migration](db/migrate/20240517100503_add_service_reference_to_appointments.rb)
3. Delete unused [file](app/javascript/controllers/hello_controller.js)

On a final note, please refrain from large commits. The smaller the better so it's easier to review and address comments. This could have been staggered between installing redis, creating reviews, setting up the different models, etc. This took a while so thanks for your patience on this.

**Pull request merged**